autorest/preparer.go

```diff
@@ -183,6 +183,16 @@ func WithBaseURL(baseURL string) PrepareDecorator {
 	}
 }
 
+// WithCustomBaseURL returns a PrepareDecorator that replaces brace-enclosed keys within the
+// request base URL (i.e., http.Request.URL) with the corresponding values from the passed map.
+func WithCustomBaseURL(baseURL string, urlParameters map[string]interface{}) PrepareDecorator {
```

Comment:
It is possible to gen `map[string]string` and modify go-autorest to take them instead of `map[string]interface{}`. The same goes for `WithEscapedPathParameters()`, `WithPathParameters()` and `WithQueryParameters()`. What do you guys think about changing those too? I could update this PR and add a new one for AutoRest tomorrow.
cc @marstr